test/unit/models/classes/Security/DataAccess/Repository/CacheKeyChainRepositoryTest.php

```diff
@@ -0,0 +1,185 @@
+                [CachedPlatformKeyChainRepository::PRIVATE_PREFIX . self::IDENTIFIER],
+                [CachedPlatformKeyChainRepository::PUBLIC_PREFIX . self::IDENTIFIER]
+            )
+            ->willReturnOnConsecutiveCalls(
+                false,
+                false
+            );
+
+        $this->keyChainGeneratorMock
+            ->expects($this->once())
+            ->method('getKeyChain')
+            ->willReturn($this->getKeyChain());
+
+        $this->simpleCacheMock
+            ->expects($this->exactly(2))
+            ->method('set')
+            ->withConsecutive(
+                [CachedPlatformKeyChainRepository::PRIVATE_PREFIX . self::IDENTIFIER, $keyChain->getPrivateKey()],
+                [CachedPlatformKeyChainRepository::PUBLIC_PREFIX . self::IDENTIFIER, $keyChain->getPublicKey()]
+            );
+
+
+        $this->simpleCacheMock
+            ->expects($this->exactly(2))
+            ->method('get')
+            ->withConsecutive(
+                [CachedPlatformKeyChainRepository::PUBLIC_PREFIX . self::IDENTIFIER],
+                [CachedPlatformKeyChainRepository::PRIVATE_PREFIX . self::IDENTIFIER]
+            )->willReturnOnConsecutiveCalls(
+                'publicKey',
+                'privateKey'
+            );
+
+        $this->subject->findAll($keyChainQuery);
+    }
+
+    public function testFindAll(): void
+    {
+        $keyChainQuery = new KeyChainQuery(self::IDENTIFIER);
+
+        $this->simpleCacheMock
+            ->method('has')
+            ->withConsecutive(
+                [CachedPlatformKeyChainRepository::PRIVATE_PREFIX . self::IDENTIFIER],
+                [CachedPlatformKeyChainRepository::PUBLIC_PREFIX . self::IDENTIFIER]
+            )
+            ->willReturnOnConsecutiveCalls(
+                true,
+                true
+            );
+
+        $this->simpleCacheMock
+            ->expects($this->exactly(2))
+            ->method('get')
+            ->withConsecutive(
+                [CachedPlatformKeyChainRepository::PUBLIC_PREFIX . self::IDENTIFIER],
+                [CachedPlatformKeyChainRepository::PRIVATE_PREFIX . self::IDENTIFIER]
+            )->willReturnOnConsecutiveCalls(
+                'publicKey',
+                'privateKey'
+            );
+
+        $result = $this->subject->findAll($keyChainQuery);
+
+        $keyChainCollection = $result->getKeyChains();
+        $this->assertSame(self::IDENTIFIER, $keyChainCollection[0]->getIdentifier());
+        $this->assertSame('privateKey', $keyChainCollection[0]->getPrivateKey()->getValue());
+        $this->assertSame('publicKey', $keyChainCollection[0]->getPublicKey()->getValue());
+    }
+
+    private function getKeyChain(): KeyChain
+    {
+        $publicKey = new Key('publicKey');
+        $privateKey = new Key('privateKey');
+
+        $keyChain = new KeyChain(
+            self::IDENTIFIER,
+            'name',
+            $publicKey,
+            $privateKey
+        );
+
+        return $keyChain;
```

Comment:
```suggestion
        return new KeyChain(
            self::IDENTIFIER,
            'name',
            new Key('publicKey'),
            new Key('privateKey')
        );
```

Saves reading time :)